src/arm_linux.rs

```diff
@@ -54,13 +56,43 @@ fn insert_aligned(aligned: u32, val: u32, shift: u32, mask: u32) -> u32 {
     (aligned & !(mask << shift)) | ((val & mask) << shift)
 }
 
+/// Loads the value at `ptr`.
+///
+/// # Safety
+///
+/// `ptr` must be aligned and point to memory within a page that allows read access.
+/// If `T` has a size of 4, then `ptr` must be valid for an atomic read.
```

Comment:
```suggestion
/// If `T` has a size of 4, then `ptr` must be valid for a relaxed atomic read.
```
However, I am a bit confused about what extra requirements you have in mind here.

Reply:
That sounds like an implementation detail of the function.

I'd expect the safety contract to be something like
- `ptr` must be 4-aligned (currently it just says "aligned" but it is ambiguous how aligned)
- `T` must have a size of at most 4
- `ptr` must be valid for a relaxed atomic read of size `size_of::<T>()`

Whether and when it uses atomics vs inline asm does not have to be exposed in the docs.